x-pack/platform/plugins/shared/onechat/server/services/tools/utils/create_internal_registry.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  ExecutableTool,
+  Runner,
+  ToolProviderGetOptions,
+  ToolProviderHasOptions,
+  ToolProviderListOptions,
+} from '@kbn/onechat-server';
+import {
+  InternalToolRegistry,
+  InternalToolProvider,
+  ScopedPublicToolRegistryFactoryFn,
+  PublicToolRegistry,
+} from '../types';
+import { combineToolProviders } from './combine_tool_providers';
+import { toExecutableTool } from './tool_conversion';
+
+export const createInternalRegistry = ({
```

Comment:
to double check my understanding:

- there are multiple Tool providers. 
- First examples of these providers are the builtIn and x MCP provided tools
- there is an internalToolRegistry which combines all providers into a single point for list, get, execute
- internal is there for more of a private name?